### PR TITLE
Convert usage telemeter to use MetricsTree

### LINKED ISSUE #1023: Use MetricsTree in usage telemeter
Depends on https://github.com/linkerd/linkerd/issues/934

The usage telemeter should pull metrics from the MetricsTree provided as a stack param instead of from the global commonMetrics registry.

### PR DESCRIPTION
**Pull request by adleong**
Fixes #1023

### CODE REVIEW

File: telemetry/core/src/main/scala/io/buoyant/telemetry/MetricsTree.scala [hunk 1]
```diff
@@ -12,6 +12,19 @@ trait MetricsTree {
...
+  }
+  def gauge: Option[Metric.Gauge] = metric match {
+    case g: Metric.Gauge => Some(g)
+    case _ => None
+  }
+  def stat: Option[Metric.Stat] = metric match {
+    case s: Metric.Stat => Some(s)
+    case _ => None
+  }
+
```
Comment (olix0r) [inline telemetry/core/src/main/scala/io/buoyant/telemetry/MetricsTree.scala]: I don't _love_ adding these to the public interface for just the one use...

Comment (olix0r) [inline telemetry/core/src/main/scala/io/buoyant/telemetry/MetricsTree.scala]: specifically, i don't like encouraging an allocation just for the sake of convenience. I'd rather callers make that decision explicitly.

Comment (adleong) [inline telemetry/core/src/main/scala/io/buoyant/telemetry/MetricsTree.scala]: the allocation of the PF?  How would you feel about this if the PFs were val'd up?

Comment (adleong) [inline telemetry/core/src/main/scala/io/buoyant/telemetry/MetricsTree.scala]: or the allocation of the Option?

Comment (olix0r) [inline telemetry/core/src/main/scala/io/buoyant/telemetry/MetricsTree.scala]: well, the PF should def be val'd... i meant the option --  usually this is just an intermediary allocation like

`t.metric match {
  case c: Metric.Counter =>
    // do the thing
  case _ =>
    // don't do it
}
```

By providing `counter`, we encourage callers to do a needless option-allocation.  it's not a huge deal, i just tend to prefer public APIs that are minimal and don't encourage undesirable patterns

**Pull request merged**